lib/codecov.js

```diff
@@ -159,10 +161,26 @@ var upload = function(args, on_success, on_failure){
 ' \\_____\\___/ \\__,_|\\___|\\___\\___/ \\_/  \n' +
 '                                ' + version);
 
+  var token = (args.options.token || process.env.codecov_token || process.env.CODECOV_TOKEN);
+
+  if (token) {
+    query.token = token;
+  }
+
   query.yaml = ['codecov.yml', '.codecov.yml'].reduce(function (result, file) {
     return result || (fs.existsSync(path.resolve(process.cwd(), file)) ? file : undefined)
   }, undefined)
 
+  // TODO: we should set other values here too based off yaml file
+  // if a yaml file was provided, then parse it and look for the token (gh #23)
+  if (query.yaml) {
+    var yaml = YAML.load(path.resolve(process.cwd(), query.yaml));
+    if (yaml.codecov && yaml.codecov.token)
+      query.token = yaml.codecov.token;
+    else
+      throw new Error('yaml file detected but `codecov` block missing');
```

Comment:
Can you say this in another way?  Confused

On Jul 25, 2016 8:57 AM, "Ed Moore" <notifications@github.com> wrote:

> In lib/codecov.js
> <https://github.com/codecov/codecov-node/pull/25#discussion_r72058217>:
>
> >    query.yaml = ['codecov.yml', '.codecov.yml'].reduce(function (result, file) {
> >      return result || (fs.existsSync(path.resolve(process.cwd(), file)) ? file : undefined)
> >    }, undefined)
> >
> > +  // TODO: we should set other values here too based off yaml file
> > +  // if a yaml file was provided, then parse it and look for the token (gh #23)
> > +  if (query.yaml) {
> > +    var yaml = YAML.load(path.resolve(process.cwd(), query.yaml));
> > +    if (yaml.codecov && yaml.codecov.token)
> > +      query.token = yaml.codecov.token;
> > +    else
> > +      throw new Error('yaml file detected but `codecov` block missing');
>
> If the codecov block is in the yaml file but the token is missing then
> this line will run as well.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/codecov/codecov-node/pull/25/files/0aeab66b6d3f2f53b1e8b0282ae56271a2664f82#r72058217>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAf7hXKcZKl1-2QkItjUFcEHXEtekriiks5qZLKugaJpZM4JUCSL>
> .
>